src/java.base/share/classes/java/lang/Class.java

```diff
@@ -438,8 +438,10 @@ private static Class<?> forName(String className, Class<?> caller)
      * obtain any of the {@code Class} objects representing primitive
      * types or void.
      *
-     * <p> If {@code name} denotes an array class, the component type of
-     * the array class is loaded but not initialized.
+     * <p> If {@code name} denotes an array class, i.e. the name consists of
+     * one or more {@code '['} followed by the binary name of the element type
+     * and {@code ';'}, the element type of the array class is loaded but
+     * not initialized.
```

Comment:
Do arrays of primitive types need special treatment?  For example, "[I"

Reply:
Thanks for catching it.  Updated.